src/utils/winlnkfileparse.cpp

```diff
@@ -0,0 +1,147 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// SPDX-FileCopyrightText: 2017-2019 Alejandro Sirgo Rica & Contributors
+
+#include "winlnkfileparse.h"
+#include <QDir>
+#include <QDirIterator>
+#include <QFileSystemModel>
+#include <QImageWriter>
+#include <QRegularExpression>
+#include <QSettings>
+#include <QString>
+
+#include <shlobj.h>
+
+WinLnkFileParser::WinLnkFileParser()
+{
+    QStringList sListImgFileExt;
+    for (const auto &ext : QImageWriter::supportedImageFormats()) {
+        sListImgFileExt.append(ext);
+    }
+    this->getImageFileextAssociates(sListImgFileExt);
+}
+
+DesktopAppData WinLnkFileParser::parseLnkFile(const QFileInfo& fiLnk,
+                                              bool& ok) const
+{
+    DesktopAppData res;
+    ok = true;
+
+    QFileInfo fiSymlink(fiLnk.symLinkTarget());
+    if (!fiSymlink.exists() ||
+            !fiSymlink.fileName().endsWith(".exe") ||
+            fiSymlink.baseName().contains("unins")) {
+        ok = false;
+        return res;
+    }
+
+    res.name = fiLnk.baseName();
+    res.exec = fiSymlink.absoluteFilePath();
+
+    // Get icon from exe
+    QFileSystemModel *model = new QFileSystemModel;
+    model->setRootPath(fiSymlink.path());
+    res.icon = model->fileIcon(model->index(fiSymlink.filePath()));
+
+    if (m_GraphicAppsList.contains(fiSymlink.fileName())) {
+        res.categories = QStringList() << "Graphics";
+    } else {
+        res.categories = QStringList() << "Utility";
+    }
+
+    for (const auto &app : m_appList) {
+        if (app.exec == res.exec) {
+            ok = false;
+            break;
+        }
+    }
+
+    if (res.exec.isEmpty() || res.name.isEmpty()) {
+        ok = false;
+    }
+    return res;
+}
+
+int WinLnkFileParser::processDirectory(const QDir& dir)
+{
+    QStringList sListMenuFilter;
+    sListMenuFilter << "Accessibility" << "Administrative Tools" << "Setup" << "System Tools" << "Uninstall" << "Update" << "Updater" << "Windows PowerShell";
+    const QString sMenuFilter("\\b(" + sListMenuFilter.join('|') + ")\\b");
+    QRegularExpression regexfilter(sMenuFilter);
+
+    bool ok;
+    int length = m_appList.length();
+    // Go through all subfolders and *.lnk files
+    QDirIterator it(dir.absolutePath(), {"*.lnk"}, QDir::NoFilter, QDirIterator::Subdirectories);
+    while (it.hasNext()) {
+        QFileInfo fiLnk(it.next());
+        if (!regexfilter.match(fiLnk.absoluteFilePath()).hasMatch()) {
+            DesktopAppData app = parseLnkFile(fiLnk, ok);
+            if (ok) {
+                m_appList.append(app);
+            }
+        }
+    }
+
+    return m_appList.length() - length;
```

Comment:
Why the subtraction here?

Reply:
To be honest: I don't know the exact reason for this return value! I know it returns the number of found applications in the folder, but the return value is (currently) not evaluated (see below). But since I didn't know, if the number of apps could be useful in future, I kept it.

I copied the existing DesktopFileParser and to keep compatibility I used the same return value
https://github.com/flameshot-org/flameshot/blob/57c8e8d893669cd83d8b279ea3ff6db50b16aa48/src/utils/desktopfileparse.cpp#L120-L123

In current applauncherwidget the return value is not processed.
https://github.com/flameshot-org/flameshot/blob/57c8e8d893669cd83d8b279ea3ff6db50b16aa48/src/tools/launcher/applauncherwidget.cpp#L47-L49